### ISSUE TITLE
Compose for Web: Copy & paste support for TextField

### ISSUE THREAD

**Issue by StefanOltmann** [JetBrains/compose-multiplatform#4036]
On the desktop version, there is a context menu offering options such as "cut", "copy", "paste" and "select all" complemented by convenient shortcuts like CMD+C and CMD+V.

It appears that a similar functionality is currently absent in the Compose for Web interface.

**Comment by terrakok** [comment]
Yes, we are going to support it in the future 👍

**Comment by usert-4** [comment]
Hello, @terrakok as we are in the subject of text selection, what about browser search browser feature ? for example on Chrome i guess it's text based on html, what about when ui is in a canvas ?
could image analyzing/almost ia help here for example for identifying text but it would be on browser side ? and others thing such as when content is out of screen/not drawn, or there are others possible ways for making this work

**Comment by eymar** [comment]
> what about text search browser feature
if this feature becomes available at some point, it probably won't work same way it works in Chrome with html pages.

It's more likely to work like in google docs, and I'd say it's a feature that an app implementation should take care of. 

Currently, it's not planned, but can be considered.

**Comment by usert-4** [comment]
yes, we could also imagine a system where Compose place html tag informations on the elements in the canvas(location etc..),  i also thinked about such thing for debugging/testing purpose(same question if 2 text are same and nothing differentiate them etc..), but that's almost another subject that could be seen later, thanks

**Comment by dhakehurst** [comment]
any idea on the timescale...when will this get supported ?

**Comment by eymar** [comment]
> any idea on the timescale...when will this get supported ?

We aim to make it properly implemented in 1.7 version of Compose Multiplatform.

**Comment by MahmoudMabrok** [comment]
I think paste not working in web version , I must enter all data, is this issue solved or included in current issue ?

**Comment by dbaroncelli** [comment]
yes, it's very annoying that cut/paste doesn't work on compose web

**Comment by usert-4** [comment]
yes @Skaldebane also semantics could help as they provide much informations, could be seen on web side how it/if used

**Comment by tombaileywzd** [comment]
> We aim to make it properly implemented in 1.7 version of Compose Multiplatform.

Just wondering if Hiragana/Katakana support is in scope for the release with cut/copy/paste support? At the moment, Hiragana and Katakana inputs on MacOS show up as Romaji. `h + e` should become `へ` but ends up as `he`

**Comment by dmentx** [comment]
Does someone have a workaround?

**Comment by Skaldebane** [comment]
@Schahen I think this relates to supporting input methods in general. Browsers tell input methods that they do support them (which they do in normal HTML), and so Compose Web has to integrate with that somehow.

Interestingly, Compose Desktop doesn't show the intermediate text and suggestions inline, but it works nevertheless. I may be totally wrong, but I assume this is because Compose Desktop doesn't announce that it supports IMEs at all, and so they work in this sort of fallback mode where the text is prepared in a small system pop-up before inputting the final result into the app. Just a wild guess, I really have no idea how IMEs work, but that's how it behaves on desktop (Linux). This is kind of a separate thing but I just thought it was interesting.

Flutter Web (wasm) seems to support this, and can be tested in any text field in the Material 3 demo: https://flutterweb-wasm.web.app/. There seems to be no Japanese font so it all ends up as tofu (the little squares), but the IME is clearly working and replacing characters inline. Maybe you can draw from their implementation.

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-4036">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
